src/qibolab/instruments/rfsoc/convert.py

```diff
@@ -55,8 +55,9 @@ def pulse_lo_frequency(pulse: Pulse, qubits: dict[int, Qubit]) -> int:
 
 def convert_units_sweeper(
     sweeper: rfsoc.Sweeper, sequence: PulseSequence, qubits: dict[int, Qubit]
-):
+) -> rfsoc.Sweeper:
     """Convert units for `qibosoq.abstract.Sweeper` considering also LOs."""
+    sweeper = rfsoc.Sweeper(**asdict(sweeper))
```

Comment:
This should be fully equivalent to `copy.deepcopy()`. I'd favor this second implementation, because it makes the purpose more explicit :)